acceptance-tests/src/test/java/uk/gov/di/test/step_definitions/Login.java

```diff
@@ -304,4 +304,38 @@ public void theUserEntersTheSixDigitSecurityCodeFromTheirEmail() {
         }
         findAndClickContinue();
     }
+
+    @When("the user enters their {string} code")
+    @When("the new user enters their {string} code")
+    public void theUserEntersTheirAuthCode(String codeType){
+
+        if (codeType == "email") {
```

Comment:
this would be better as an if/else or a switch statement. for example, if it is 'email' then why bother to run the next 'if' statement

Reply:
There's also an assumption here that the tester will enter 'text message' in lower case. What about if they enter 'Text message'? This would fail. Have a look at '.toLowerCase()'

Reply:
I personally prefer to have individual if statements than if/else. I always found it easier when dubugging my programming that I could see the specific if block the code fell over on.

Reply:
It's about coding standards and writing efficient code. That way might be your preference but you aren't the only person that will be using/reading/approving this code. Please change to if/else-if, or use switch.
Also, my 2nd comment re. .toLowerCase() hasn't been addressed.
Happy to run through it on a call if you have any queries?